### PR TITLE
fix: use post for searchRuns/use accurate multitrial detection

### PR DESCRIPTION
**Pull request by ashtonG**
## Description
this fixes a few issues with the multitrial check for run actions

* when the selection grew too large, the resulting url triggered a 431 error. we're switching the method of the search runs endpoint to post in order to get around this.
* the multitrial check used the wrong method to check if a run was part of a search
* the warning shown if the multitrial check returned positive had incorrect copy

## Test Plan
* navigate to the runs table
* select all runs in the current page
* select the move action
* the check should succeed and the notice should show that multitrial runs will be moved along with other runs in their search

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: performance/src/api_performance_tests.ts [hunk 1]
```diff
@@ -336,23 +333,19 @@ const getloadTests = (
...
       "search flat runs w/ hparam",
-      getRequest(
-        `/api/v1/runs?filter=%7B%22filterGroup%22%3A%7B%22children` +
-        `%22%3A%5B%7B%22columnName%22%3A%22hp.global_batch_size%22%2C%22kind` +
-        `%22%3A%22field%22%2C%22location%22%3A%22LOCATION_TYPE_RUN_HYPERPARAMETERS` + 
-        `%22%2C%22operator%22%3A%22%3E%22%2C%22type%22%3A%22COLUMN_TYPE_NUMBER` +
-        `%22%2C%22value%22%3A0.1%7D%5D%2C%22conjunction%22%3A%22and%22%2C%22kind` +
-        `%22%3A%22group%22%7D%2C%22showArchived%22%3Afalse%7D`,
-      ),
+      testRequest('POST', '/api/v1/runs', JSON.stringify({
```
Comment (AmanuelAaron) [inline performance/src/api_performance_tests.ts:342]: I think we should run the perf test once to confirm that this is working. I think it should.

**Review by AmanuelAaron — APPROVED** [review]
Backend LGTM

File: performance/src/api_performance_tests.ts (hunk 1 above)
Comment (ashtonG) [reply performance/src/api_performance_tests.ts:342]: ran it -- needed to stringify the filter param. rerunning now.

**Comment by keita-determined** [comment]
can we fix the conflicts? i'll review it afterwards

**Review by keita-determined — COMMENTED** [review]
i got this error on this branch. i deleted all DB and tested in on the incognito mode
Do i miss something?

image

File: webui/react/src/components/RunFilterInterstitialModalComponent.tsx [hunk 2]
```diff
@@ -126,7 +126,7 @@ export const RunFilterInterstitialModalComponent = forwardRef<ControlledModalRef
           const results = await searchRuns(
             {
               filter: JSON.stringify(filter),
-              limit: 0,
+              limit: -2,
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply webui/react/src/components/RunFilterInterstitialModalComponent.tsx:129]: `-2` returns empty results, leaving just the pagination.

File: performance/src/utils/helpers.ts [hunk 3]
```diff
@@ -46,24 +46,22 @@ export const authenticateVU = (clusterURL: string): string => {
...
-};
-
+  testConfig?: TestConfiguration
+) => <RT extends ResponseType | undefined>(method: string, url: string, body?: RequestBody, params?: RefinedParams<RT>): () => void => () => {
+    const { headers, ...restParams } = params || {};
+    const response = http.request(method, generateEndpointUrl(url, clusterURL), body, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: testConfig?.auth.token || '',
+        ...headers || {}
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply performance/src/utils/helpers.ts:58]: the parentheses appear to be unnecessary here -- try copying the following in a js console.
```jsx
let foo
console.log({bar: "baz", ...foo || {}}) // should print { bar: "baz" }
foo = { bar: "qux" };
console.log({ bar: "baz, ...foo || {}}) // should print { bar: "qux" }
```

**Comment by ashtonG** [comment]
@keita-determined fixed -- i did a little code golfing before committing which didn't work out.

File: performance/src/utils/helpers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (keita-determined) [reply performance/src/utils/helpers.ts:58]: didnt know that. JS behavior is so weird

**Review by keita-determined — APPROVED** [review]
working good with many experiments

---
out of scope: do we want it in the experiment table as well, or just flat run since we eventually remove the experiment table?

**Comment by ashtonG** [comment]
we do, but I want to bundle that work with leveraging some other things we can do to reduce the backend complexity now that the parameters don't have to be strings. here's the ticket: https://hpe-aiatscale.atlassian.net/browse/ET-602

**Pull request merged**